### PR TITLE
13.0 sale packaging min sellable qty

### PR DESCRIPTION
**Pull request by mmequignon**
This PR adds a computed minimum sellable qty field on `product.template` and `product.product`

### CODE REVIEW

File: sale_by_packaging/models/product_product.py [hunk 1]
```diff
@@ -1,13 +1,35 @@
...
+            "if Only Sell by Packaging is set."
+        ),
+    )
+
+    @api.depends(
+        "sell_only_by_packaging", "uom_id", "packaging_ids", "packaging_ids.qty"
+    )
+    def _compute_variant_min_sellable_qty(self):
+        for record in self:
+            record.min_sellable_qty = False
```
Comment (TDu) [inline sale_by_packaging/models/product_product.py]:
```suggestion
            record.min_sellable_qty = 0
```
It is a float so...

File: sale_by_packaging/models/product_template.py [hunk 2]
```diff
@@ -14,6 +14,32 @@ class ProductTemplate(models.Model):
...
+            "if Only Sell by Packaging is set."
+        ),
+    )
+
+    @api.depends(
+        "sell_only_by_packaging",
+        "uom_id",
+        "packaging_ids",
+        "packaging_ids.qty",
+        "product_variant_ids",
```
Comment (TDu) [inline sale_by_packaging/models/product_template.py]: Maybe also adding
`"product_variant_ids.min_sellable_qty"`

**Review by TDu — COMMENTED** [review]
It would be nice to have this covered by tests, also I think they are broken...

File: sale_by_packaging/models/product_product.py [hunk 3]
```diff
@@ -1,13 +1,38 @@
...
+            "Minimum sellable quantity, according to the available packagings, "
+            "if Only Sell by Packaging is set."
+        ),
+    )
+
+    @api.depends(
+        "sell_only_by_packaging",
+        "uom_id.factor",
+        "packaging_ids.qty",
+        "min_sellable_qty",
```
Comment (sebalix) [inline sale_by_packaging/models/product_product.py]: `min_sellable_qty` depends on `min_sellable_qty`? To remove I guess?

File: sale_by_packaging/models/product_template.py [hunk 4]
```diff
@@ -14,6 +14,31 @@ class ProductTemplate(models.Model):
...
+        help=(
+            "Minimum sellable quantity, according to the available packagings, "
+            "if Only Sell by Packaging is set."
+        ),
+    )
+
+    @api.depends(
+        "sell_only_by_packaging",
+        "uom_id.factor",
+        "product_variant_ids",
```
Comment (sebalix) [inline sale_by_packaging/models/product_template.py]:
```suggestion
```
Already included through `product_variant_ids.min_sellable_qty`

**Review by sebalix — APPROVED** [review]

File: sale_by_packaging/models/product_product.py [hunk 5]
```diff
@@ -1,13 +1,35 @@
...
+    )
+
+    @api.depends(
+        "sell_only_by_packaging", "uom_id.factor", "packaging_ids.qty",
+    )
+    def _compute_variant_min_sellable_qty(self):
+        for record in self:
+            record.min_sellable_qty = 0.0
+            if record.sell_only_by_packaging and record.packaging_ids:
+                record.min_sellable_qty = fields.first(
```
Comment (simahawk) [inline sale_by_packaging/models/product_product.py:31]: seems you are missing to filter packaging on `can_be_sold`

**Review by simahawk — CHANGES_REQUESTED** [review]

File: sale_by_packaging/models/product_product.py [hunk 6]
```diff
@@ -1,13 +1,35 @@
...
+                    record.packaging_ids.sorted(lambda p: p.qty)
+                ).qty
+            elif not record.sell_only_by_packaging:
+                record.min_sellable_qty = record.uom_id.factor
```
Comment (rousseldenis) [inline sale_by_packaging/models/product_product.py:35]: Why this ?

Comment (sebalix) [reply sale_by_packaging/models/product_product.py:35]: ping @mmequignon

Comment (TDu) [reply sale_by_packaging/models/product_product.py:35]: In my opinion, this `elif` should just be removed.
And also the `min_sellable_qty` field should only be displayed in the view if the product is set with `sell_only_by_packaging` set to True

File: sale_by_packaging/models/product_product.py (hunk 5 above)
Comment (TDu) [reply sale_by_packaging/models/product_product.py:31]: Has been fixed on a later push

File: sale_by_packaging/models/product_product.py (hunk 6 above)
Comment (TDu) [reply sale_by_packaging/models/product_product.py:35]: I implemented those changes on the forward port  to 14 #1784
I will bring them here if they are good.

**Review by sebalix — CHANGES_REQUESTED** [review]
Commit to reword `sale_by_packaging: Add mininmum sellable qty on product` to be inline with the ported one in https://github.com/OCA/sale-workflow/pull/1784/commits (and to remove the version from it)

**Pull request closed without merging**